### DISCUSSION TITLE
Unit tests failing on my machine? (... length of 6 but got 5)

### DISCUSSION THREAD

**Discussion by ghornich** [jeemok/better-npm-audit#68] [Q&A] [answered]
Hello,

I'm planning to contribute to better-npm-audit, but having trouble with running the unit tests:

#getExceptionsIds
      √ should display the vulnerabilities from command line if .nsprc file not given
      **1) should combine the exceptions from command line and .nsprc file**
    #processExceptions
      **2) should be able to process exceptions correctly
      3) should be able to filter active exceptions and label correctly
      4) should be able to filter inactive exceptions and label correctly
      5) should be able to filter expired exceptions and label correctly
      6) should be able to filter invalid exceptions and label correctly**

All the issues look like this. **AssertionError: expected [ 985, 1213, 2000, 2001, 2100 ] to have a length of 6 but got 5**

Environment: Win10, VS Code. I tried with multiple node/npm versions, clean workspace, WSL, nothing works. Any ideas?

Thanks!

**Comment by jeemok** [comment, marked as answer]
Hello @ghornich,

Thanks for your interest in wanting to contribute, I'd really need some hands for this project.

Are these unit tests failing before you made any code changes, or after? if you have made some code changes, could you create a PR and I can help you from there

**Reply by ghornich** [reply]
Hi @jeemok!

I reverted all my changes, and still getting the errors. I tried on 2 different windows machines, and on my Ubuntu laptop. Clean download, npm i. Tried with different node and npm versions (node 14.3,0, 15.3.0, 16.3.2, npm 6.x, 7.0.14, 8.x).

Still getting these test fails :( The same 6 tests. I've attached the npm logs.

[npmtest.txt](https://github.com/jeemok/better-npm-audit/files/7997532/npmtest.txt)

**Reply by jeemok** [reply]
hey @ghornich, sorry for the late reply! I've checked the logs and saw the error `Failed parsing .nsprc file: SyntaxError: Unexpected end of JSON input`, could it be `.nsprc` file not in proper JSON format? i.e. having a trailing comma at the end

if you could provide the `.nsprc` file maybe I can help to take a look too

**Reply by ghornich** [reply]
Hey @jeemok! I finally figured it out: if I changed the date of `1654` in the nsprc file from the original 2021-12-31 date to 2022-12-31, the tests work! I think the tests shouldn't rely on the current date, but some stub date.

...but @alexburkowskypolysign was faster. submitted a test fix in `e7dab5`. So I guess this thread is solved. Thanks!

**Reply by jeemok** [reply]
definitely should use a stub date! Glad to solve this mystery @ghornich!
